### PR TITLE
v2.0.0 Project rewritten for better usage

### PR DESCRIPTION
**Pull request by DatL4g**
## This project is rewritten so it requires a new major release as it contains breaking changes and improvements

# Kotlin DSL

The new implementation uses the Kotlin DSL feature with type-safe builders.
https://kotlinlang.org/docs/type-safe-builders.html#how-it-works

## What does that mean?

An instance of StatusBarAlert can be created as follows:

```kotlin
statusBarAlert {
    autoHide()
    duration(2, TimeUnit.SECONDS)
    text("Kotlin DSL")
    showProgress(false)
    alertColor(android.R.color.red)
    textColor(android.R.color.white)
}
```

The old creating method (mainly intended for Java) is still usable, however method names changed.

## What else changed?

To improve the DSL usage, most method names changed e.g. ```withText("text")``` became ```text("text")```
**View the Builder class for more details.**

### Show and hide methods

In the previous version, creating an instance was a pain because it got shown immediatly after building.
This changed, so you have call ```.show()``` to display the StatusBarAlert after building.

The old hide method didn't make sense at all, because it was designed to handle multiple StatusBarAlerts although only one is shown.
So when showing a new one now the old one gets hidden as before and the new one shows up after that.
The ```.hide()``` method gets called on the instance now instead of the messy static workaround method.

### Overall: The old behavior remains but with improved usage

### CODE REVIEW

**Comment by fede87** [comment]
Thank you for the huge improvement!

File: README.MD [hunk 1]
```diff
@@ -26,69 +26,66 @@ This lib is supported by every device with a SDK level of at least 14 (Android 4
...
+Create a new StatusBarAlert in activities and fragments.
 
-```
+Using Kotlin DSL:
 
-### Quick Sample
+```kotlin
+statusBarAlert {
+    autoHide()
+    dduration(2, TimeUnit.SECONDS)
```
Comment (DerTyp7214) [inline README.MD:54]: shouldnt this be `duration`?

Comment (DatL4g) [reply README.MD:54]: Yes sorry misspelled it in the README, however in code it's correctly

Comment (fede87) [reply README.MD:54]: I've fixed the readme, thank you

**Pull request merged**